app/chats/mutations/createAdminMessage.tsx

```diff
@@ -1,17 +1,22 @@
 import db from "db"
-import { Ctx } from "blitz"
+import { Ctx, resolver } from "blitz"
 import { newMessageMailer } from "emails/newMessageMailer"
 import { dbLogger as log } from "app/lib/logger"
+import { Message } from "../validation"
 
-export default async function createAdminMessage({ content, chatId }, context: Ctx) {
-  context.session.$authorize()
-  await db.adminMessage.create({
-    data: { content: content, sentIn: { connect: { id: chatId } } },
-  })
-  await sentMail(chatId, content)
+export default resolver.pipe(
+  resolver.zod(Message),
+  resolver.authorize(),
+  async ({ content, chatId }, context: Ctx) => {
+    context.session.$authorize()
```

Comment:
Why do we have to call `context.session.$authorize()` if we called `resolver.authorize()` before.

Reply:
See https://github.com/blitz-js/blitz/discussions/2472